### ISSUE TITLE
PUT not supported

### ISSUE THREAD

**Issue by oddobert** [tvdsluijs/python_ddns#2]
ddns - ERROR - /zones.dns_records.put DOMAIN.DOMAIN - 10000 PUT method not allowed for the api_token authentication scheme - api call failed

(Domain.Domain no the actual domain).

Changes leads to new error.
PUT changed to post in line 126 gives 
2021-11-07 19:55:53,587 - ddns - ERROR - /zones.dns_records. - 9207 Content-type must be application/json. - api call failed

**Comment by oddobert** [comment]
This one is possibly because self.new_dns_record is None.
Investigating further

**Comment by oddobert** [comment]
Yes, this is a logical error, where the set_cloud_dns fails if there ip address hasn't changed. - it wont create new_dns_records, but it will still return true. 
Adding a else: return False statement in between 118 and 119 will take care of that .

**Comment by tvdsluijs** [comment]
Thank you Oddobert for investigating this. I will take a look at this.

**Comment by tvdsluijs** [comment]
> Yes, this is a logical error, where the set_cloud_dns fails if there ip address hasn't changed. - it wont create new_dns_records, but it will still return true.
> Adding a else: return False statement for the if(self.current_ip != self.cloud_flare_ip) will fix that.

I'm not so sure if this helps. 

I do a
```
if self.cloud_flare_ip is not None and self.cloud_flare_ip == self.current_ip:
            return True
```

right after getting the DNS info. If that checks out, the script will wait another 15 minutes as there seems no changes are needed.
